### DISCUSSION TITLE
Integration with Compose / Multi Key

### DISCUSSION THREAD

**Discussion by dufferzafar** [espanso/espanso#1175] [Feature requests and Ideas]
Unix has the concept of a [Compose Key](https://en.wikipedia.org/wiki/Compose_key), [Debian Docs](https://wiki.debian.org/XCompose).

You first set the compose key - I have mine set to CapsLock.

Then you can get espanso like functionality by defining a rule in the `~/.XCompose` file:

```
<Multi_key> <n> : "Shadab Zafar"
```

Now when you press `CapsLock` followed by `n` you get "Shadab Zafar". A disadvantage of using the Compose functionality is that you don't actually see any of the characters being typed.

In espanso, I would achieve similar things by a config like:

```
matches:
  - trigger: ";n"
    replace: "Shadab Zafar"
```

I was wondering if espanso could use the Compose Key allowing us to specify triggers that start with the compose key. So, instead of specifying a trigger like `;n` I could use `<ComposeKey>n`.

**Comment by federico-terzi** [comment]
Hey @dufferzafar,

This is very interesting indeed! I didn't know about the existence of such functionality on Linux :)
I'll make a few experiments and let you know once I know something!

Do you have any idea of where the actual <Multi_key> mapping is stored? I'm talking about the mapping between <Multi_key> = CapsLock

Cheers :)

**Comment by dufferzafar** [comment]
I've been using Linux for 5-6 years now, and I found out about Compose yesterday from the HN thread about AutoHotkey: https://news.ycombinator.com/item?id=23156060

I'm not aware of where this mapping is stored. I made the change from KDE settings:

![image](https://user-images.githubusercontent.com/1449512/81949037-feba5b80-961f-11ea-8bf1-b8ee8280b36c.png)

**Comment by federico-terzi** [comment]
All right, thanks for the heads up @dufferzafar

I'll take a closer look as soon as I get my hands on my Linux machine :)
